likecoin/js/admin/likecoin_metabox.js

```diff
@@ -56,35 +56,127 @@ async function onISCNCallback(event) {
   ISCNTextField.innerHTML = `<a rel="noopener" target="_blank" href="${url}">${status}</a>`;
 }
 
+async function uploadToArweave(data) {
+  try {
+    const { tx_hash: txHash, error, success } = data;
+    if (error || success === false) return;
+    const res = await jQuery.ajax({
+      type: 'POST',
+      url: `${wpApiSettings.root}likecoin/v1/posts/${wpApiSettings.postId}/arweave/upload`,
+      dataType: 'json',
+      contentType: 'application/json; charset=UTF-8',
+      data: JSON.stringify({ txHash }), // LIKEpay txHash
+      method: 'POST',
+      beforeSend: (xhr) => {
+        xhr.setRequestHeader('X-WP-Nonce', wpApiSettings.nonce);
+      },
+    });
+    if (!res.data || !res.data.arweaveId) {
+      throw new Error('NO_ARWEAVE_ID_RETURNED'); // Could be insufficient fund or other error.
+    }
+  } catch (error) {
+    console.error(`Error occurs when uploading to Arweave: ${error}`);
+  }
+}
+
 function onSubmitToISCN(e) {
-  e.preventDefault();
   const {
-    title,
-    ipfsHash,
-    tags,
-    url,
+    title, mattersIPFSHash, arweaveIPFSHash, tags, url, arweaveId,
   } = lcPostInfo;
   const { siteurl } = wpApiSettings;
-  if (!ipfsHash) return;
-  const titleString = encodeURIComponent(title);
-  const tagsArray = tags || [];
-  const tagsString = tagsArray.join(',');
-  const urlString = encodeURIComponent(url);
-  const redirectString = encodeURIComponent(siteurl);
-  const likeCoISCNWidget = `https://like.co/in/widget/iscn?fingerprint=${ipfsHash}&publisher=matters&title=${titleString}&tags=${tagsString}&opener=1&url=${urlString}&redirect_uri=${redirectString}`;
-  window.open(likeCoISCNWidget, '_blank', 'menubar=no,location=no,width=576,height=768');
-  window.addEventListener('message', onISCNCallback, false);
+  try {
+    if (!mattersIPFSHash && !arweaveIPFSHash && !arweaveId) {
+      throw new Error('NO_IPFS_HASH_NOR_ARWEAVE_ID_FOUND');
+    }
+    const titleString = encodeURIComponent(title);
+    const tagsArray = tags || [];
+    const tagsString = tagsArray.join(',');
+    const urlString = encodeURIComponent(url);
+    const redirectString = encodeURIComponent(siteurl);
+    let fingerprint = '';
+    if (mattersIPFSHash) {
+      const mattersIPFSHashFingerprint = `ipfs://${mattersIPFSHash}`;
+      if (fingerprint) {
+        fingerprint = fingerprint.concat(`,${mattersIPFSHashFingerprint}`);
```

Comment:
I think `concat()` is not preferred for string operation
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/concat#performance
just use `+=` or template string